lib/processes/visTestPattern.cpp

```diff
@@ -38,71 +51,41 @@ visTestPattern::visTestPattern(Config& config, const std::string& unique_name,
-            throw std::invalid_argument("fakeVis: length of frequencies array ("
-                                        + std::to_string(bins.size())
-                                        + ") can not be larger "
-                                          "than num_freq ("
-                                        + std::to_string(num_freq) + ").");
-        }
+    _tolerance = config.get_default<float>(unique_name, "tolerance", 1e-6);
+    _report_freq = config.get_default<uint64_t>(unique_name, "report_freq", 1000);
 
-        exp_val_freq = std::vector<cfloat>(num_freq);
-        for (size_t i = 0; i < num_freq; i++) {
-            size_t j;
-            for (j = 0; j < bins.size(); j++) {
-                if (bins.at(j) == i)
-                    break;
-            }
-            if (j == bins.size())
-                exp_val_freq[i] = default_val;
-            else
-                exp_val_freq[i] = bin_values.at(j);
+    if (_tolerance < 0)
+        throw std::invalid_argument("visCheckTestPattern: tolerance has to be positive (is "
+                                    + std::to_string(_tolerance) + ").");
+
+    write_dir = config.get<std::string>(unique_name, "write_dir");
+    if (opendir(write_dir.c_str()) == nullptr) {
+        // Create directory
+        if (mkdir(write_dir.c_str(), S_IRWXU | S_IRGRP | S_IROTH) < 0) {
+            std::string error =
+                fmt::format("Failure creating directory {}: {}", write_dir, std::strerror(errno));
+            throw std::runtime_error(error);
         }
-    } else
-        throw std::invalid_argument("visCheckTestpattern: unknown mode: " + mode);
+    }
+    INFO("Writing report to '%s'.", write_dir.c_str());
 
-    tolerance = config.get_default<float>(unique_name, "tolerance", 1e-6);
-    report_freq = config.get_default<uint64_t>(unique_name, "report_freq", 1000);
+    endpoint_name = config.get_default<std::string>(unique_name, "endpoint_name", "run_test");
 
-    outfile_name = config.get<std::string>(unique_name, "out_file");
+    // Don't run any tests until update received.
+    num_frames = 0;
 
-    if (tolerance < 0)
-        throw std::invalid_argument("visCheckTestPattern: tolerance has to be"
-                                    " positive (is "
-                                    + std::to_string(tolerance) + ").");
+    expected_data_ready = false;
 
-    outfile.open(outfile_name);
-    if (!outfile.is_open()) {
-        throw std::ios_base::failure("visCheckTestPattern: Failed to open "
-                                     "out file "
-                                     + outfile_name);
-    }
-    outfile << "fpga_count,time,freq_id,num_bad,avg_err,min_err,max_err" << std::endl;
+    // Subscribe to the dynamic config update: used to start a test for a number of frames.
+    kotekan::restServer::instance().register_post_callback(
+        endpoint_name, std::bind(&visTestPattern::receive_update, this, std::placeholders::_1,
```

Comment:
There should be a `remove_json_callback` in the destructor for this endpoint.

Reply:
Thanks!
Should we rename `register_post_callback` `register_json_callback` or is there a reason in the naming difference between the `register_` and `remove_` methods?

Reply:
We should rename them, I'm not sure why we ended up with different names.  I think both should be `*_json_callback` since it's not a generic POST callback, it does assume a JSON encoded string.